util/dnsname/dnsname.go

```diff
@@ -17,3 +17,11 @@ func HasSuffix(name, suffix string) bool {
 	nameBase := strings.TrimSuffix(name, suffix)
 	return len(nameBase) < len(name) && strings.HasSuffix(nameBase, ".")
 }
+
+// SanitizeHostname returns hostname with some common suffixes removed.
+func SanitizeHostname(hostname string) string {
+	hostname = strings.TrimSuffix(hostname, ".local")
+	hostname = strings.TrimSuffix(hostname, ".localdomain")
+	hostname = strings.TrimSuffix(hostname, ".lan")
+	return hostname
```

Comment:
We could probably add even more sanitation here (to mirror the labeling sanitation done in corp https://github.com/tailscale/corp/blob/b6dd5c7a2b91c80e0f3973e01c21b2c03e59c444/control/dnsutil/name.go#L117)

Reply:
Ah I thought I'd seen this stuff before! Feel free to move it to oss and deduplicate.

Reply:
Sounds good, will do!